Sources/Plasma/CoreLib/hsStream.cpp

```diff
@@ -703,26 +708,26 @@ void hsUNIXStream::Rewind()
         return;
     fBytesRead = 0;
     fPosition = 0;
-    (void)::fseek(fRef, 0, SEEK_SET);
+    fseek(fRef, 0, SEEK_SET);
 }
 
 void hsUNIXStream::FastFwd()
 {
     if (!fRef)
         return;
-    (void)::fseek(fRef, 0, SEEK_END);
+    fseek(fRef, 0, SEEK_END);
     fBytesRead = fPosition = ftell(fRef);
 }
 
-uint32_t  hsUNIXStream::GetEOF()
+size_t  hsUNIXStream::GetEOF()
 {
     if( !fRef )
         return 0;
 
-    long oldPos = ftell( fRef );
-    (void)::fseek( fRef, 0, SEEK_END );
-    uint32_t end = (uint32_t)ftell( fRef );
-    (void)::fseek( fRef, oldPos, SEEK_SET );
+    size_t oldPos = ftell(fRef);
+    fseek(fRef, 0, SEEK_END);
+    size_t end = ftell(fRef);
+    fseek(fRef, oldPos, SEEK_SET);
```

Comment:
`ftell` and `fseek` actually use `long` values...  This can be cast in the return statement.

Reply:
So more generally, how do we want to handle position and EOF?

On *nix 32-bit, `long` and `size_t` are the same size (4-bytes)
On Windows 32-bit, `long` and `size_t` are the same size (4-bytes)
On *nix 64-bit, `long` and `size_t` are the same size (8-bytes)
On Windows 64-bit, `long` and `size_t` are **not** the same size (4 bytes and 8 bytes)

If we type `fPosition` as `long`, then we maybe avoid some of the concerns Hoikas identified on Windows 64-bit, but those concerns would apply on Linux 64-bit.

If we type `fPosition` as `size_t` (which this PR currently does), then we open the can of worms of potentially serializing 8-byte positions instead of 4-byte positions, as well as needing to use `_fseeki64`/`_ftelli64` on Windows.

Reply:
My comment was more about the signedness than the size (`size_t` is unsigned), but the size is a valid concern too...  Although in this particular function, it only matters once we start dealing with files >2GiB.

Reply:
I'd prefer for us to avoid using types that are different sizes on different platforms.

Reply:
There's no perfect way to handle this, but if we want to fully support large files (even on 32-bit systems), you'll need to extend the workarounds a bit further.  For example:

```c++
#ifdef _MSC_VER
    int64_t oldPos = _ftelli64(fRef);
    _fseeki64(fRef, 0, SEEK_END);
    int64_t end = _ftelli64(fRef);
    _fseeki64(fRef, oldPos, SEEK_SET);
#else
    // With -D_FILE_OFFSET_BITS=64
    static_assert(sizeof(off_t) >= sizeof(int64_t));
    off_t oldPos = ftello(fRef);
    fseeko(fRef, 0, SEEK_END);
    off_t end = ftello(fRef);
    fseeko(fRef, oldPos, SEEK_SET);
#endif
    return static_cast<int64_t>(end);
```